### ISSUE TITLE
Unable to mount multiple handlers to the same prefix

### ISSUE THREAD

**Issue by olefasting** [google/dart-neats#136]
This was originally posted as an issue with shelf, but I guess this is the right place for this:

Say that I have two services (an authentication service and a profiles service, in my case) and an API service that is meant to bind it all together into a final handler. If I want to prefix the two services with an /api' prefix, by mounting both to /api/ in the API service, only the first handler will be accessible when running the server.

I have tested that everything works, otherwise, by mounting the two services to different prefixes, so the issue seems to lie in the fact that the mounts are both to the same prefix...

This seems like a bug to me, as it is normal practice in libraries like express, which shelf seems to emulate...

Also, when I remap the authentication service to / and keep the profiles service at /api/, the latter will only work if it is mapped before the former. Very strange behavior, if you ask me....

```dart
class AuthenticationService {
  AuthenticationService();

  Handler get handler {
    final router = Router();

    /* Various handlers mapped to /login, /register etc */

    return router;
  }
}

class ProfileService {
  ProfileService();

  Handler get handler {
    final router = Router();

    /* Various handlers, all mapped to /profile, but with different methods */

    return router;
  }
}

class ApiService {
  Handler get handler {
    final router = Router();

    final authenticationService = AuthenticationService();
    final profileService = ProfileService();

    router.mount('/api/', profileService.handler); // Only this will work. If I change the order, it is always the first mount that works....
    router.mount('/api/', authenticationService.handler);

    return router;
  }
}
```

**Comment by jonasfj** [comment]
I think we did consider implementing the semantics from [Cascade](https://pub.dev/documentation/shelf/latest/shelf/Cascade-class.html), but then it wouldn't be possible to return a custom 404 for a custom route, and have a different catch-all 404 for all routes.

For now consider using [Cascade](https://pub.dev/documentation/shelf/latest/shelf/Cascade-class.html) for mounts under `api/`.

I do see the problem, but this would be a breaking change, please have people vote up if they think this is desired behavior.

**Comment by jonasfj** [comment]
Let's introduce a sentinel `final routeNotFound = shelf.Response.notFound('route not found');`, and whenever a router has no matching route it returns the `routeNotFound` sentinel object.

Whenver a router gets a response from a handler or sub-router it will check if response is the sentinel (`response == routeNotFound`) if this is the case then the response is ignored and other routes are matched.

To do:
 *  [ ] Add sentinel
 *  [ ] Update documentation for router and example
 *  [ ] Add tests
 *  [ ] Update `CHANGELOG.md` explaining who default behavior for nested routers have changed, and why we chose not to make this a breaking change.

**Comment by jonasfj** [comment]
Fixed in `shelf_router` version `1.1.0`.
